src/scripts/main.js

```diff
@@ -1,3 +1,64 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('.growth-table');
+  const appendRowButton = document.querySelector('.append-row');
+  const removeRowButton = document.querySelector('.remove-row');
+  const appendColumnButton = document.querySelector('.append-column');
+  const removeColumnButton = document.querySelector('.remove-column');
+
+  const maxCount = 10;
+  const minCount = 2;
+
+  const updateButtons = () => {
+    const rowCount = table.rows.length;
+    const colCount = table.rows[0].cells.length;
+
+    appendRowButton.disabled = rowCount >= maxCount;
+    removeRowButton.disabled = rowCount <= minCount;
+    appendColumnButton.disabled = colCount >= maxCount;
+    removeColumnButton.disabled = colCount <= minCount;
+  };
+
+  const appendRow = () => {
+    const newRow = table.insertRow();
+    const colCount = table.rows[0].cells.length;
```

Comment:
Similar to the previous comment, ensure that the table has at least one row initially to avoid errors when accessing `table.rows[0]`. Consider adding a check to ensure the table is not empty before accessing its rows.